### PR TITLE
fix(eslint-plugin): [no-unnecessary-boolean-literal-compare] fixer should handle parentheses

### LINKED ISSUE #6567: Bug: [no-unnecessary-boolean-literal-compare] Quick Fix generates bug with `instanceof`
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3gDwNwFgAoIgSwDMAKbeE9DKFMEJM+AURhiRngF5-4ZKBDQgAlPADeAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ARhw5IhTWo3yJoQ+LTIcAtsRGJ0UAO4imkcGAC+IM0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "warn"
  }
}
```

### Repro Code

```typescript
declare const x;

if(x instanceof Error === false) {}
```

is turned into

```ts
declare const x;

if(!x instanceof Error) {}
```

which is semantically different:

Operator precedence: `===` < `instanceof` < `!`

### Expected Result

Firstly, there should be an option to explicitly allow `(identifier instanceof identifier === false)`.\
I prefer it over `(!(identifier instanceof identifier))`.

As for the actual bug, the following code is expected (with the aforementioned example and the above option disabled):
```ts
declare const x;

if(!(x instanceof Error)) {}
```

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6567
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Add missing handling for removal and adding parenthesis in rule fixer

new cases covered:

```ts
if ((x instanceof Error) === false) {}
if (false === (x instanceof Error)) {}
if (x instanceof Error === false) {}
```

old fixer result

```ts
if (!(x instanceof Error) { }
if (!x instanceof Error)) { }
if (!x instanceof Error) { }
```

new fixer result

```ts
if (!(x instanceof Error)) { }
if (!(x instanceof Error)) { }
if (!(x instanceof Error)) { }
```

--------------

initially i was planning to extend range of fixer to include `)(` but after further investigation this  started generating to many edge cases

eg. `(false) === x instanceof Error`

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts [hunk 1]
```diff
@@ -227,7 +223,10 @@ export default util.createRule<Options, MessageIds>({
 
         context.report({
           fix: function* (fixer) {
```
Comment (armano2) [inline packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts:225]: after further investigation we could simplify this fixer to

```ts
          fix: function* (fixer) {
            const isUnaryNegation =
              node.parent != null && nodeIsUnaryNegation(node.parent);

            const mutatedNode = isUnaryNegation ? node.parent! : node;

            yield fixer.replaceText(
              mutatedNode,
              sourceCode.getText(comparison.expression),
            );

            const shouldNegate = comparison.literalBooleanInComparison
              ? !comparison.negated
              : comparison.negated;

            if (isUnaryNegation === shouldNegate) {
              yield fixer.insertTextBefore(mutatedNode, '!');

              if (!util.isStrongPrecedenceNode(comparison.expression)) {
                yield fixer.insertTextBefore(mutatedNode, '(');
                yield fixer.insertTextAfter(mutatedNode, ')');
              }
            }

            // if the expression `exp` is nullable, and we're not comparing to `true`,
            // we can just replace the entire comparison with `exp` or `!exp`
            if (
              comparison.expressionIsNullableBoolean &&
              !comparison.literalBooleanInComparison
            ) {
              // provide the default `true`
              yield fixer.insertTextBefore(mutatedNode, '(');
              yield fixer.insertTextAfter(mutatedNode, ' ?? true)');
            }
          }
```

that should allow us to provide better fixer for

```ts
## source
if (!(varBoolean === false)) {}
## current
if (!(!varBoolean)) {}
## new
if (varBoolean) {}
```

i'm not sure if we should push this with this fix

File: packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts [hunk 2]
```diff
@@ -227,7 +223,10 @@ export default util.createRule<Options, MessageIds>({
...
-            yield fixer.removeRange(comparison.range);
+            yield fixer.replaceText(
+              node,
+              sourceCode.getText(comparison.expression),
+            );
 
             // if the expression `exp` isn't nullable, or we're comparing to `true`,
             // we can just replace the entire comparison with `exp` or `!exp`
```
Comment (armano2) [inline packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts:232]: after further investigation we could simplify this fixer to

```ts
          fix: function* (fixer) {
            const isUnaryNegation =
              node.parent != null && nodeIsUnaryNegation(node.parent);

            const mutatedNode = isUnaryNegation ? node.parent! : node;

            yield fixer.replaceText(
              mutatedNode,
              sourceCode.getText(comparison.expression),
            );

            const shouldNegate = comparison.literalBooleanInComparison
              ? !comparison.negated
              : comparison.negated;

            if (isUnaryNegation === shouldNegate) {
              yield fixer.insertTextBefore(mutatedNode, '!');

              if (!util.isStrongPrecedenceNode(comparison.expression)) {
                yield fixer.insertTextBefore(mutatedNode, '(');
                yield fixer.insertTextAfter(mutatedNode, ')');
              }
            }

            // if the expression `exp` is nullable, and we're not comparing to `true`,
            // we can just replace the entire comparison with `exp` or `!exp`
            if (
              comparison.expressionIsNullableBoolean &&
              !comparison.literalBooleanInComparison
            ) {
              // provide the default `true`
              yield fixer.insertTextBefore(mutatedNode, '(');
              yield fixer.insertTextAfter(mutatedNode, ' ?? true)');
            }
          }
```

that should allow us to provide better fixer for

```ts
## source
if (!(varBoolean === false)) {}
## current
if (!(!varBoolean)) {}
## new
if (varBoolean) {}
```

i'm not sure if we should push this with this fix

Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts:232]: I like it 😄 +1 to including it here. Also a +1 to filing a followup if you want to get this in sooner.

**Review by JoshuaKGoldberg — APPROVED** [review]
💯

File: packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts (hunk 2 above)
Comment (bradzacher) [reply packages/eslint-plugin/src/rules/no-unnecessary-boolean-literal-compare.ts:232]: I'm going to land this as-is for now, just so we can get this fix out with the next release.
We can follow-up next week with the changes to simplify this fixer later

**Pull request merged**